src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs

```diff
@@ -2800,15 +2800,16 @@ private void GenerateHelperFunctions(TextWriter writer)
 
```

Comment:
@vexx32, your last commit had 10 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Tests for (error, warning, etc) action preference.Setting ErrorActionPreference to stop prevents user from getting the error exception.$err.Count
```powershell
Expected 1, but got 2.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Language/Scripting/ActionPreference.Tests.ps1: line 79
79:         It '$err.Count' { $err.Count | Should -Be 1 }
```
Test PSDesiredStateConfiguration.Invoke-DscResource.mof resources.Set method should work
```powershell
System.InvalidOperationException: Failed to install the module 'PsDscResources'.
 ---> System.Management.Automation.RuntimeException: Cannot index into a null array.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception) in /Users/runner/runners/2.168.2/work/1/s/src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs:line 1654
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame) in /Users/runner/runners/2.168.2/work/1/s/src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs:line 504
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in /Users/runner/runners/2.168.2/work/1/s/src/System.Management.Automation/engine/interpreter/ControlFlowInstructions.cs:line 391
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in /Users/runner/runners/2.168.2/work/1/s/src/System.Management.Automation/engine/interpreter/ControlFlowInstructions.cs:line 357
   --- End of inner exception stack trace ---
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/a/bins/publish/Modules/PowerShellGet/Modules/PowerShellGet.LocalizationHelper/PowerShellGet.LocalizationHelper.psm1: line 85
85:     throw $errorRecordToThrow

```
Test PSDesiredStateConfiguration.Invoke-DscResource.mof resources.Test method should return false
```powershell
unexpected state - no resources found - get-dscresource should have thrown
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/a/bins/publish/Modules/PSDesiredStateConfiguration/PSDesiredStateConfiguration.psm1: line 4685
4685:         throw "unexpected state - no resources found - get-dscresource should have thrown"

```
Test PSDesiredStateConfiguration.Invoke-DscResource.mof resources.Test method should return true
```powershell
unexpected state - no resources found - get-dscresource should have thrown
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/a/bins/publish/Modules/PSDesiredStateConfiguration/PSDesiredStateConfiguration.psm1: line 4685
4685:         throw "unexpected state - no resources found - get-dscresource should have thrown"

```
Test PSDesiredStateConfiguration.Invoke-DscResource.mof resources.Test method should return true with moduleSpecification
```powershell
You cannot call a method on a null-valued expression.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/PSDesiredStateConfiguration/PSDesiredStateConfiguration.Tests.ps1: line 462
462:                 $moduleSpecification = @{ModuleName = $module.Name; ModuleVersion = $module.Version.ToString() }

```